### PR TITLE
Fix heatmap row highlighting

### PR DESCRIPTION
**Pull request by evansd**
Fixes a couple of row highlighting bugs with a hack, followed by a terrible hack.

### CODE REVIEW

File: apps/heatmap.py [hunk 1]
```diff
@@ -148,8 +148,16 @@ def make_highlight_rect(y_index):
     if col_name == 'result_category':
         highlight_entities = set(map(int, highlight_entities))
 
+    # Worse hack: there's some weird off-by-one error in the highlighting which
+    # only manifests when grouping by test code over all tests codes. The
+    # result is that everything works as expected except that the row above the
+    # selected row gets highlighted. This "fixes" it.
```
Comment (sebbacon) [inline apps/heatmap.py]: Is it worth looking at this together or do you feel pretty confident we'll just both lose the will to live? I can live with this for now if so!

Comment (evansd) [reply apps/heatmap.py]: I feel like I've spent too much time on this already. It's not a straightfoward logic bug as it doesn't manifest if you group by test with some subset of test codes, only when you have all test codes.

Comment (sebbacon) [reply apps/heatmap.py]: Fair enough - I wonder if it's worth mentioning that thing about subsets for anyone ever returning to this?

Comment (evansd) [reply apps/heatmap.py]: Yes, good point. I've expanded the comment

**Pull request merged**